iroha/src/wsv.rs

```diff
@@ -72,45 +72,80 @@ impl WorldStateView {
 
     /// Get all `Domain`s without an ability to modify them.
     pub fn read_all_domains(&self) -> Vec<&Domain> {
-        self.world.domains.values().collect()
+        let mut vec = self.world.domains.values().collect::<Vec<&Domain>>();
+        vec.sort();
+        vec
```

Comment:
I don't think this sorts by ID. We either need to modify `PartialOrd` and `Ord` implementations in the data model crate. Or sort explicitly by id here.

It is important to sort by id, so that it can be easily replicated in client libs in different languages (Java, etc.). Also we need to check the Ord and PartialOrd impls for ids, so that it can be easily explained to client developers, maybe add a doc comment for this.

Also please add a note in all the WSV functions here, that they return sorted content by ids.

Reply:
This comment relates to all WSV functions that we have here.

Reply:
Actually Domain doesn't have id inside, rather it is identified by name. I guess we can derive `Ord` here as it will check by name first.